packages/medusa/src/repositories/product-category.ts

```diff
@@ -1,20 +1,44 @@
-import { EntityRepository, TreeRepository, Brackets, ILike } from "typeorm"
```

Comment:
Would appreciate a deeper look into this one. I didn't quite like how it ended up looking. Any ideas on how this can be made better will be really helpful. 

Need some comments on this one:

Goal:

- To separately be able to filter children and parent tree nodes with default scope
- To separately be able to filter only on category items (default scope + filter params)

Note that these are 2 different cases we needed to handle. Check line 54 and and line 47

Reply:
for the children you can look into `createDescendantsQueryBuilder` and apply the constraint for the children on that

Reply:
So you have a query builder for the top level category with the constraints etc, then you create a `createDescendantsQueryBuilder` where you apply child constraints etc. let me know

Reply:
Thanks @adrien2p. I've fixed the issue with the scopes in the new commit. `createDescendantsQueryBuilder` doesn't help me with the 2 goals I have. Take a look at the new commit.